### DISCUSSION TITLE
Setting the 'default' action on table views.

### DISCUSSION THREAD

**Discussion by perrelet** [filamentphp/filament#2059] [Help] [answered]
By default, tables use the route of the ViewRecord as the URL in cell anchor links (if one exists). 

How would one override this, so that when a cell is clicked, the action is the EditRecord's route?

i.e Looking to make the edit action 'default', whilst retaining the view action as an option on the right hand side of the table.

Tx

**Comment by danharrin** [comment, marked as answer]
Try this, on the List page class:

```php
protected function getTableRecordUrlUsing(): ?Closure
{
    return function (Model $record): ?string {
        $resource = static::getResource();

        if (! $resource::canEdit($record)) {
            return null;
        }

        return $resource::getUrl('edit', ['record' => $record]);
    };
}
```

**Comment by perrelet** [comment]
@danharrin Works a treat - thanks kindly for the support!

**Comment by Abdulrehman3802** [comment]
`->recordUrl(fn (Model $record): string => EditPage::getUrl(['record' => $record]))`
 
 add it in table after column it is one line solution.
- Make sure to replace EditPage with you Resource Page

![image](https://github.com/user-attachments/assets/7bbd31df-b7a1-498f-9f99-ad1295095bb6)
